plugins/obs-outputs/WebRTCStream.cpp

```diff
@@ -688,15 +688,36 @@ webrtc::VideoFrame WebRTCStream::constructOutputFrame(video_data* frame)
 
 	buffer = i444Buffer;
     } else {
-        videoType = webrtc::VideoType::kNV12;
-        uint32_t size = outputWidth * outputHeight * 3 / 2;
-        int stride_y = outputWidth;
-        int stride_uv = (outputWidth + 1) / 2;
+
+	uint8_t* frameData = frame->data[0];
+	std::unique_ptr<uint8_t[]> convertedFrameData;
+
+	switch (videoFormat) {
+	case VIDEO_FORMAT_NV12:
+	    videoType = webrtc::VideoType::kNV12;
+	    break;
+
+	case VIDEO_FORMAT_I420:
+	    videoType = webrtc::VideoType::kI420;
+	    break;
+	case VIDEO_FORMAT_RGBA:
+	    videoType = webrtc::VideoType::kARGB;
+	    unsigned int frameSizeBytes = frame->linesize[0] * outputHeight * 4;
+	    convertedFrameData.reset(new uint8_t[frameSizeBytes]);
+	    libyuv::ARGBToABGR(frameData, frame->linesize[0], convertedFrameData.get(), frame->linesize[0], outputWidth, outputHeight);
```

Comment:
Is this supposed to be converting to ABGR rather than RGBA?

Reply:
Yes, I found it interesting too.

I've checked the color byte order in frame data, and it's `ABGR` for `VIDEO_FORMAT_RGBA`.
That's why applying `libyuv::ARGBToABGR` worked - it just flipped it back from `ABGR` to `ARGB`.

Hope It's just a naming issue, and not some endianness issue.

Reply:
Yeah strange.  Only other item I saw that I wondered about was `valid_format` functions in various plugins that allow through I420, I444, and NV12, but not RGB.  Haven't done the investigation to see if this is relevant.  If not, happy to see this merge; lmk if I need to approve or if you're good to go.

Reply:
I believe it's a plugin specific thing for some internal functionality. 
Example in `obs-ffmpeg-vaapi.c` it's:

```cpp
static inline bool valid_format(enum video_format format)
{
	return format == VIDEO_FORMAT_NV12;
}
```

Also, in the settings (as described in the ticket) we cat select all those video formats (`I420`, `I444`, `NV12`, `RGB`) and we do receive frames (in the method) in those formats.